v2/auth_test.go

```diff
@@ -374,19 +382,28 @@ func TestDirectProvider(t *testing.T) {
 }
 
 func TestDirectProvider_WithCustomUserIDFunc(t *testing.T) {
-	_, teardown := prepService(t)
+	_, teardown := prepService(t, func(svc *Service) {
+		svc.AddDirectProviderWithUserIDFunc("directCustom",
```

Comment:
Note that using a provider name with `_`  symbol breaks some expectations in [Authenticator.isProviderAllowed()](https://github.com/go-pkgz/auth/blob/master/v2/middleware/auth.go#L156-L167) method.

Reply:
I will add these changes to v1 as well.

Reply:
I copied same changes to `v1/auth.go` and `v1/auth_test.go` as well.
These files are same in v1 and v2 at the moment.